pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/httproute_reconciler.go

```diff
@@ -0,0 +1,804 @@
+	var err error
+	url := &knapis.URL{}
+	url.Scheme = ingressConfig.UrlScheme
+	url.Host, err = GenerateDomainName(isvc.Name, isvc.ObjectMeta, ingressConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return url, nil
+}
+
+func getRawServiceHost(isvc *v1beta1.InferenceService, client client.Client) string {
+	existingService := &corev1.Service{}
+	if isvc.Spec.Transformer != nil {
+		transformerName := constants.TransformerServiceName(isvc.Name)
+
+		// Check if existing transformer service name has default suffix
+		err := client.Get(context.TODO(), types.NamespacedName{Name: constants.DefaultTransformerServiceName(isvc.Name), Namespace: isvc.Namespace}, existingService)
+		if err == nil {
+			transformerName = constants.DefaultTransformerServiceName(isvc.Name)
+		}
+		return network.GetServiceHostname(transformerName, isvc.Namespace)
+	}
+
+	predictorName := constants.PredictorServiceName(isvc.Name)
+
+	// Check if existing predictor service name has default suffix
+	err := client.Get(context.TODO(), types.NamespacedName{Name: constants.DefaultPredictorServiceName(isvc.Name), Namespace: isvc.Namespace}, existingService)
+	if err == nil {
+		predictorName = constants.DefaultPredictorServiceName(isvc.Name)
+	}
+	return network.GetServiceHostname(predictorName, isvc.Namespace)
+}
+
+func createHTTPRouteMatch(prefix string) gatewayapiv1.HTTPRouteMatch {
+	return gatewayapiv1.HTTPRouteMatch{
+		Path: &gatewayapiv1.HTTPPathMatch{
+			Type:  utils.ToPointer(gatewayapiv1.PathMatchRegularExpression),
+			Value: utils.ToPointer(prefix),
+		},
+	}
+}
+
+func addIsvcHeaders(name string, namespace string) gatewayapiv1.HTTPRouteFilter {
+	return gatewayapiv1.HTTPRouteFilter{
+		Type: gatewayapiv1.HTTPRouteFilterRequestHeaderModifier,
+		RequestHeaderModifier: &gatewayapiv1.HTTPHeaderFilter{
+			Set: []gatewayapiv1.HTTPHeader{
+				{
+					Name:  constants.IsvcNameHeader,
+					Value: name,
+				},
+				{
+					Name:  constants.IsvcNamespaceHeader,
+					Value: namespace,
+				},
+			},
+		},
+	}
+}
+
+func createHTTPRouteRule(routeMatches []gatewayapiv1.HTTPRouteMatch, filters []gatewayapiv1.HTTPRouteFilter,
+	serviceName, namespace string, port int32) gatewayapiv1.HTTPRouteRule {
+	var backendRefs []gatewayapiv1.HTTPBackendRef
+	if serviceName != "" {
+		backendRefs = []gatewayapiv1.HTTPBackendRef{
+			{
+				BackendRef: gatewayapiv1.BackendRef{
+					BackendObjectReference: gatewayapiv1.BackendObjectReference{
+						Kind:      utils.ToPointer(gatewayapiv1.Kind(constants.ServiceKind)),
+						Name:      gatewayapiv1.ObjectName(serviceName),
+						Namespace: (*gatewayapiv1.Namespace)(&namespace),
+						Port:      (*gatewayapiv1.PortNumber)(&port),
+					},
+				},
+			},
+		}
+	}
+	return gatewayapiv1.HTTPRouteRule{
+		Matches:     routeMatches,
+		Filters:     filters,
+		BackendRefs: backendRefs,
+		Timeouts: &gatewayapiv1.HTTPRouteTimeouts{
+			Request: utils.ToPointer(gatewayapiv1.Duration("60s")),
```

Comment:
Timeout value should be getting from the field on inference service

Reply:
Done